js/core/index.js

```diff
@@ -43,3 +57,44 @@ function Runtime() {
 }
 
 global.runtime = module.exports = new Runtime();
+
+runtime.tty.stdout = new runtime.tty.StdOut();
+runtime.tty.stdin = new runtime.tty.StdIn();
+initio(runtime);
+
+runtime.stdio = {
+  stdout: runtime.tty.stdout,
+  stdin: runtime.tty.stdin,
+  StdIn: runtime.tty.StdIn,
+  StdOut: runtime.tty.StdOut,
+  createOut: function(funcs) {
+    if (funcs) {
+      funcs.onwrite = funcs.onwrite || runtime.stdio.stdout.onwrite;
+      funcs.onsetcolor = funcs.onsetcolor || runtime.stdio.stdout.onsetcolor;
+      funcs.onsetbackgroundcolor = funcs.onsetbackgroundcolor || runtime.stdio.stdout.onsetbackgroundcolor;
+    }
+
+    var newout = new runtime.stdio.StdOut();
+    newout.onwrite = funcs.onwrite;
+    newout.onsetcolor = funcs.onsetcolor;
+    newout.onsetbackgroundcolor = funcs.onsetbackgroundcolor;
+    return newout;
+  },
+  createIn: function(funcs) {
+    if (funcs) {
+      funcs.onread = funcs.onread || runtime.stdio.stdin.onread;
+      funcs.onreadline = funcs.onreadline || runtime.stdio.stdin.onreadline;
+    }
+
+    var newin = new runtime.stdio.StdIn();
+    newin.onread = funcs.onread;
+    newin.onreadline = funcs.onreadline;
+
+    return newin;
+  }
+};
+
+global.env = {
```

Comment:
probably not good to expose it as global, it's accessible as `runtime.stdio` anyway